csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -170,8 +181,21 @@ TensorView* AllocationOrderInferencer::resolveAllocationOrder(
       continue;
     }
 
+    auto iter = alloc_order_map_.find(tv);
     // skip entry that doesn't have an allocation order
-    if (alloc_order_map_.count(tv) == 0) {
+    if (iter == alloc_order_map_.end()) {
+      return nullptr;
+    }
+
+    // skip entry that has an empty allocation order
+    if (iter->second.empty()) {
```

Comment:
Will we be able to get rid of this if-then (therefore simplifies) if non_bc_high_water_mark is initialized to -inf?

Reply:
I don't think so. We still won't want to make sure factory-created tensor view yield to allocation order propagation from inputs where they have a real impact on vectorization.

i.e. initialization of non_bc_high_water_mark wouldn't change that behavior.

Reply:
> inputs where they have a real impact on vectorization.

But won't those inputs have a positive non_bc_count and therefore "dominate"? I failed to see how an all-broadcast allocation domain can affect vectorization. 

Anyhow, if you think we need to distinguish empty allocation domain differently from all-broadcast allocation domain, you can **optionally** simplify the code a little bit

```
src = nullptr;
src_loop_domain_count = -inf;
src_allocation_order_size = -inf;

for each candidate {
  skip non TensorView or those that don't exist in alloc_order_map_. 

  loop_domain_count = countLoopIterDomains(candidate);
  allocation_order_size = candidate's allocation order size;
  if (make_pair(loop_domain_count, allocation_order_size) > make_pair(src_loop_domain_count, src_allocation_order_size)) {
    src = candidate;
    src_loop_domain_count = loop_domain_count;
    src_allocation_order_size = allocation_order_size;
  }
}
```

This better clarifies the intention of picking the candidate that has the most loop IterDomains and if there's a tie prefer the largest allocation_order size. The current pattern
```
if (...) {
  if (...) {
  }
  continue;
}
```
is a bit hard to follow.

Reply:
> I failed to see how an all-broadcast allocation domain can affect vectorization.

So factory method doesn't give an all-broadcast domain. Admittedly, `full` probably should, but we can also have random operation that shouldn't do that.

Quick example here:

```
T1_l[ iS2{16}, iS3{32} ]
   = full({16, 32}, float(1));
T2_g[ iS4{16}, iS5{32} ]
   = T0_g[ iS0{i0}, bS1{1} ]
   + T1_l[ iS2{16}, iS3{32} ];

TransformPrinter :
T0_g[ iS0{i0}, bS1{1} ]
 root domain : (iS0{i0}, bS1{1})
 allocation domain : (bS1{1}, iS0{i0})
T1_l[ iS2{16}, iS3{32} ]
 root domain : (iS2{16}, iS3{32})
 allocation domain : (iS2{16}, iS3{32})
T2_g[ iS4{16}, iS5{32} ]
 root domain : (iS4{16}, iS5{32})
```

i.e. T1 here is a tensor created by factory method, it has a higher loop_domain_count, but it shouldn't propagate its allocation order to T2.
We would want T0's allocation order to propagate to T2, which would allow both IO tensor to have the same innermost dimension -> vectorized load/store on both buffer.

(this is a bad example, since even when propagation failed, transpose scheduler should kick in and still give us vectorized load/store. But I hope it carries the argument across)

Reply:
> T1 here is a tensor created by factory method, it has a higher loop_domain_count

I see -- countLoopIterDomains looks at the TensorView not `iter->second`. So the code I suggested would fail.